src/test/java/tatracker/logic/commands/student/EditStudentCommandTest.java

```diff
@@ -1,177 +1,204 @@
-    //     Student editedStudent = model.getFilteredStudentList().get(INDEX_FIRST_STUDENT.getZeroBased());
-    //
-    //     String expectedMessage = String.format(EditStudentCommand.MESSAGE_EDIT_STUDENT_SUCCESS, editedStudent);
-    //
-    //     Model expectedModel = new ModelManager(new TaTracker(model.getTaTracker()), new UserPrefs());
-    //
-    //     assertCommandSuccess(editStudentCommand, model, expectedMessage, expectedModel);
-    // }
-
-    /*
+    private Model model = new ModelManager(getTypicalTaTrackerWithStudents(), new UserPrefs());
+
+    @Test
+    public void execute_allFieldsSpecifiedUnfilteredList_success() {
+        Student editedStudent = new StudentBuilder().build();
+        EditStudentDescriptor descriptor = new EditStudentDescriptorBuilder(editedStudent).build();
+        EditStudentCommand editStudentCommand = new EditStudentCommand(MATRIC_FIRST_STUDENT,
+                getTypicalModule().getIdentifier(), getTypicalGroup().getIdentifier(), descriptor);
+
+        String expectedMessage = String.format(EditStudentCommand.MESSAGE_EDIT_STUDENT_SUCCESS,
+                editedStudent, getTypicalModule().getIdentifier(), getTypicalGroup().getIdentifier());
+
+        ModelManager expectedModel = new ModelManager(model.getTaTracker(), new UserPrefs());
+        expectedModel.setStudent(model.getFilteredStudentList().get(0), editedStudent,
+                getTypicalGroup().getIdentifier(), getTypicalModule().getIdentifier());
+
+        assertEditStudentCommandSuccess(editStudentCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_someFieldsSpecifiedUnfilteredList_success() {
+        Index indexLastStudent = Index.fromOneBased(model.getFilteredStudentList().size());
+        Student lastStudent = model.getFilteredStudentList().get(indexLastStudent.getZeroBased());
+
+        StudentBuilder studentInList = new StudentBuilder(lastStudent);
+        Student editedStudent = studentInList.withName(VALID_NAME_BOB).withPhone(VALID_PHONE_BOB)
+                .withTags(VALID_TAG_HUSBAND).build();
+
+        EditStudentDescriptor descriptor = new EditStudentDescriptorBuilder().withName(VALID_NAME_BOB)
+                .withPhone(VALID_PHONE_BOB).withTags(VALID_TAG_HUSBAND).build();
+        EditStudentCommand editStudentCommand = new EditStudentCommand(lastStudent.getMatric(),
+                getTypicalModule().getIdentifier(), getTypicalGroup().getIdentifier(), descriptor);
+
+        String expectedMessage = String.format(EditStudentCommand.MESSAGE_EDIT_STUDENT_SUCCESS,
+                editedStudent, getTypicalModule().getIdentifier(), getTypicalGroup().getIdentifier());
+
+        ModelManager expectedModel = new ModelManager(model.getTaTracker(), new UserPrefs());
+        expectedModel.setStudent(lastStudent, editedStudent);
+
+        assertEditStudentCommandSuccess(editStudentCommand, model, expectedMessage, expectedModel);
+    }
```

Comment:
Err I don't think the test case should be using indices. Should be with matric numbers?

Reply:
this is just to get the last student in the list tho, editing is still done with matric numbers